core/src/main/java/org/mule/util/ClassUtils.java

```diff
@@ -1065,4 +1067,70 @@ public static URL getClassPathRoot(Class clazz)
             return annotation.getClass();
         }
     }
+
+    /**
+     * Checks that {@code value} is an instance of {@code type}. The value
+     * that this method adds over something like {@link Class#isInstance(Object)}
+     * is that it also considers the case in which {@code type} and {@code value}
+     * are evaluate by {@link #isWrapperAndPrimitivePair(Class, Class)} as {@code true}
+     *
+     * @param type  the {@link Class} you want to check the {@code value against}
+     * @param value an instance you want to verify is instance of {@code type}
+     * @param <T>   the generic type of {@code type}
+     * @return {@code true} if {@code value} is an instance of {@code type} or if they are a wrapper-primitive pair.
+     *          {@code false} otherwise
+     */
+    public static <T> boolean isInstance(Class<T> type, Object value) {
+        if (value == null) {
+            return false;
+        }
+
+        if (type.isInstance(value)) {
+            return true;
+        }
+
+        Class<?> valueType = value.getClass();
+        return isWrapperAndPrimitivePair(type, valueType) || isWrapperAndPrimitivePair(valueType, type);
+    }
+
+    /**
+     * Checks that a wrapper-primitive relationship exists between the two types, no matter which one
+     * is the wrapper or which is the primitive. For example, this method will return {@code true} for both
+     * the {@link Double}/{@code double} and the {@code double}/{@link Double} pairs. Notice that
+     * {@code false} will be returned for the pair {@link Long}/{code int} since they don't represent
+     * the same data type.
+     * <p/>
+     * If any of the two types is neither wrappers or primitives, it will return {@code false}
+     *
+     * @param type1 a {@link Class} presumed to be a wrapper or primitive type related to {@code type2}
+     * @param type2 a {@link Class} presumed to be a wrapper or primitive type related to {@code type1}
+     * @return {@code true} if the types are a wrapper/primitive pair referring to the same data type. {@code} otherwise.
+     */
+    public static boolean isWrapperAndPrimitivePair(Class<?> type1, Class<?> type2)
```

Comment:
right now no, but since this is an utils class, I don't see what's the problem exposing it